src/test/java/com/google/devtools/build/lib/starlark/StarlarkDefinedAspectsTest.java

```diff
@@ -8418,6 +8418,32 @@ public void testRuleAspectWithMandatoryParameterProvided() throws Exception {
     assertThat(aspectAResult).isEqualTo("aspect_a on target //test:dep_target, p = p_v1");
   }
 
+  @Test
+  public void testAspectLabelIsRepoMapped() throws Exception {
+    scratch.appendFile("WORKSPACE", "workspace(name = 'my_repo')");
+    scratch.file(
+        "test/aspect.bzl",
+        "load(':rule.bzl', 'MyInfo')",
+        "def _impl(target, ctx):",
+        "   if MyInfo not in target:",
+        "       fail('Provider identity mismatch')",
+        "   return struct()",
+        "MyAspect = aspect(implementation=_impl)");
+    scratch.file(
+        "test/rule.bzl",
+        "MyInfo = provider()",
+        "def _impl(ctx):",
+        "    return [MyInfo()]",
+        "my_rule = rule(implementation=_impl)");
+    scratch.file("test/BUILD",
+        "load(':rule.bzl', 'my_rule')",
+        "my_rule(name = 'target')");
+
+    AnalysisResult result = update(ImmutableList.of("@my_repo//test:aspect.bzl%MyAspect"),
```

Comment:
Hmm, what repo mapping is this label going through? How would this test fail without your change?

Reply:
I manually verified that it fails without my change.

It goes through the main repositories repo mapping, which AFAIU maps the workspace name `my_repo` to the canonical name, which is just the empty string. Before my change, the label would not go through repo mapping at all, which would cause the `MyInfo` provider to be loaded from the (non-existent) `my_repo` repo and thus differ from the one defined in the main repo.

Reply:
I see, in a real WORKSPACE, there will be a local_repository called "my_repo" that pointing to the main project's root directory to make that "external repo" available, which is an unfortunate hack, but it seems that's not set up for this test case which makes this test case work. Thanks for the clarifying!